test/CadlRanchProjects/inheritance/inheritance.tsp

```diff
@@ -42,6 +42,18 @@ model Salmon extends Fish {
   partner?: Fish;
 }
 
+@doc("Test duplicated discriminator declaration, see issue 3504")
+@discriminator("kind")
+model Turtle {
+    kind: string;
+}
+
+@doc("See turtle")
+model SeaTurtle extends Turtle {
+    kind: "seeTurtle";
+    swimmingSpeed: int32;
+}
```

Comment:
Could we have another case where the discriminator property has a type of user-defined enumeration?
Something like this:
```
enum FishType {
	Salmon: "salmon";
// etc
}
@discriminator("kind)
model Fish {
	kind: FishType;
}

model Salmon extends Fish {
	kind: FishType.Salmon;
}
```
It is OK not to add this if typespec cannot support this correctly. I suggest this because swagger can do this.

Reply:
Since the compiler does not accept the above syntax, it is fine that we do not have it.

Reply:
It's not supported. File https://github.com/Azure/autorest.csharp/issues/3561 to track it.